backend/src/index.ts

```diff
@@ -296,6 +309,79 @@ app.get("/user", async (req, res: Response) => {
   });
 });
 
+app.post("/society/create", async (req:TypedRequest<CreateSocietyBody>, res: Response) => {
+  const society = req.body;
+  if (!society.name || !society.userId) {
+    return res.status(400).json({ message: 'Invalid input.' });
+  }
+
+  if (!society.profilePicture) {
+    society.profilePicture = "null";
+  }
+
+  const sessionFromDB = await validateSession(req.session ? req.session : null);
+  if (!sessionFromDB) {
+    return res.status(401).json({ message: 'Invalid session provided.' });
+  }
+  
+  const newSociety = await prisma.society.create({
+    data: {
+      name: society.name,
+      admin: {
+        connect: {
+          id: society.userId,
+        }
+      }
+    }
+  })
+
+  return res.status(200).json(newSociety);
+})
+
+app.post("/event/create", async (req: TypedRequest<CreateEventBody>, res:Response) => {
+  //Session validation
+  const event = req.body 
+
+  const sessionFromDB = await validateSession(req.session ? req.session : null);
+  if (!sessionFromDB) {
+    return res.status(401).json({ message: 'Invalid session provided.' });
+  }
+
+  //Sanitize Inputs/Check Validity
+  if(!isValidDate(event.startDateTime, event.endDateTime)) {
+    return res.status(400).json({ message: "Invalid date"})
+  }
+  
+  //Do some db stuff TODO: RELATION LINK THING
+  const eventRes = await prisma.event.create({
+    data: {
+      banner: event.banner,
+      name: event.name,
+      startDateTime: dayjs(event.startDateTime).toISOString(),
+      endDateTime: dayjs(event.endDateTime).toISOString(),
+      location: event.location,
+      description: event.description,
+      society: {
+        connect: {
+          id: event.societyId
+        }
+      }
+    }
+  })
+  return res.status(200).json({ eventRes })
+})
+
+function isValidDate(startDate:Dayjs, endDate:Dayjs):boolean{
+  //Should probably add more test cases here
+  if (dayjs(startDate).isAfter(endDate) || dayjs(startDate).isSame(endDate) || 
+  dayjs(startDate).isBefore(dayjs())){
+    return false;
+  }
+  
+  return true;

```

Comment:
redundant return true/false just return the result of the condition:

```js
// removed boolean as it can be inferred from simplicity of function
function isValidDate(startDate: string, endDate: string) {
    let startDate = dayjs(startDate);
    ...
    return (dayjs(startDate).isAfter... );
}